indico/modules/events/editing/client/js/components/FileManager/FileList.jsx

```diff
@@ -0,0 +1,144 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2019 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React, {useContext, useCallback, useState} from 'react';
+import PropTypes from 'prop-types';
+import {useDropzone} from 'react-dropzone';
+import {Icon} from 'semantic-ui-react';
+import {FileManagerContext, filePropTypes, uploadFiles, deleteFile} from './util';
+import * as actions from './actions';
+
+import './FileManager.module.scss';
+
+function FileAction({onClick, active, icon, className}) {
+  return (
+    <Icon
+      className={className}
+      name={active ? 'spinner' : icon}
+      loading={active}
+      onClick={() => {
+        if (!active) {
+          onClick();
+        }
+      }}
+    />
+  );
+}
+
+FileAction.propTypes = {
+  onClick: PropTypes.func.isRequired,
+  active: PropTypes.bool.isRequired,
+  icon: PropTypes.string.isRequired,
+  className: PropTypes.string.isRequired,
+};
+
+function FileEntry({
+  eventId,
+  fileTypeId,
+  allowMultipleFiles,
+  file: {uuid, filename, state, claimed},
+}) {
+  const dispatch = useContext(FileManagerContext);
+  const [activeButton, setActiveButton] = useState(null);
+
+  const onDrop = useCallback(
+    async acceptedFiles => {
+      setActiveButton('replace');
+      await uploadFiles(actions.markModified, fileTypeId, acceptedFiles, eventId, dispatch, uuid);
```

Comment:
We don't need a `setActiveButton(null)` afterwards?

Reply:
No, actually React will complain about that, since the component will be unmounted by then.

Reply:
might be worth adding a comment mentioning that the component will be unmounted afterwards and thus no update is needed

Reply:
Good idea.